packages/cubejs-backend-shared/src/promises.ts

```diff
@@ -264,28 +265,35 @@ export const retryWithTimeout = <T>(
     timeout
   );
 
-/**
- * Creates a debounced version of an asynchronous function.
- */
-export const asyncDebounce = <Ret, Arguments>(
+export type AsyncDebounceOptions = {
+  max?: number;
+  ttl?: number;
+};
+
+export const asyncDebounceFn = <Ret, Arguments>(
   fn: (...args: Arguments[]) => Promise<Ret>,
+  options: AsyncDebounceOptions = {}
 ) => {
-  const cache = new Map<string, Promise<Ret>>();
+  const { max = 100, ttl = 60 * 1000 } = options;
+
+  const cache = new LRUCache<string, Promise<Ret>>({
+    max,
+    ttl,
+  });
 
   return async (...args: Arguments[]) => {
     const key = crypto.createHash('md5')
       .update(args.map((v) => JSON.stringify(v)).join(','))
       .digest('hex');
 
-    if (cache.has(key)) {
-      return <Promise<Ret>>cache.get(key);
+    const existing = cache.get(key);
+    if (existing) {
+      return existing;
     }
 
     try {
       const promise = fn(...args);
-
       cache.set(key, promise);
-
       return await promise;
```

Comment:
Should this be just `return promise;`?

Reply:
Nope, because it should clean the promise from the cache after the execution of the operation.